src/retriever/chain.ts

```diff
@@ -68,123 +69,52 @@ export class ChainRetriever implements PieceRetriever {
+        return await this.childRetriever.fetchPiece(commp, client, options)
       }
+      throw createError(
+        'ChainRetriever',
+        'fetchPiece',
+        `Failed to retrieve piece ${commp.toString()}: ${reason}`
+      )
+    }
 
-      try {
-        // Phase 1: Check if provider has the piece
-        const findUrl = constructFindPieceUrl(provider.pdpUrl, commp)
-        const findResponse = await fetch(findUrl, { signal: controller.signal })
-
-        if (!findResponse.ok) {
-          // Provider doesn't have the piece
-          failures.push({ provider: provider.owner, error: `findPiece returned ${findResponse.status}` })
-          throw new Error('Provider does not have piece')
-        }
-
-        // Phase 2: Provider has piece, download it
-        const downloadUrl = constructPieceUrl(provider.pieceRetrievalUrl, commp)
-        const response = await fetch(downloadUrl, { signal: controller.signal })
-
-        if (response.ok) {
-          // Don't cancel here! Let Promise.race decide the winner
-          return { response, index }
-        }
-
-        // Download failed
-        failures.push({ provider: provider.owner, error: `download returned ${response.status}` })
-        throw new Error(`Download failed with status ${response.status}`)
-      } catch (error: any) {
-        // Log actual failures
-        const errorMsg = error.message ?? 'Unknown error'
-        if (!failures.some(f => f.provider === provider.owner)) {
-          failures.push({ provider: provider.owner, error: errorMsg })
-        }
-        // TODO: remove this at some point, it might get noisy
-        console.warn(`Failed to fetch from provider ${provider.owner}:`, errorMsg)
-        throw error
-      }
-    })
-
+    // Step 1: Find providers
+    let providersToTry: ApprovedProviderInfo[] = []
     try {
-      // Use Promise.any to get the first successful response
-      const { response, index: winnerIndex } = await Promise.any(providerAttempts)
-
-      // Now that we have a winner, cancel all other requests
-      abortControllers.forEach((ctrl, i) => {
-        if (i !== winnerIndex) {
-          ctrl.abort()
-        }
-      })
-
-      return response
+      providersToTry = await this.findProviders(client, options?.providerAddress)
     } catch (error) {
-      // Promise.any throws AggregateError when all promises reject
-      if (error instanceof AggregateError) {
-        // All providers failed
-        const failureDetails = failures.map(f => `${f.provider}: ${f.error}`).join('; ')
-        throw createError(
-          'ChainRetriever',
-          'fetchFromProviders',
-          `All providers failed to serve piece ${commp.toString()}. Details: ${failureDetails}`
-        )
-      }
-      // Re-throw unexpected errors
-      throw error
+      // Provider discovery failed - this is a critical error
+      return await tryChildOrThrow(
+        'Provider discovery failed and no child retriever was configured'
```

Comment:
```suggestion
        'Provider discovery failed and no additional retriever method was configured'
```

"child" is an implementation detail, imagine an end-user seeing this